configuration.go

```diff
@@ -32,6 +32,23 @@ func (s ServerSuffrage) String() string {
 	return "ServerSuffrage"
 }
 
+// ConfigurationStore provides an interface that can optionally be implemented by FSMs
+// to store configuration updates made in the replicated log. In general this is only
+// necessary for FSMs that mutate durable state directly instead of applying changes
+// in memory and snapshotting periodically. By storing configuration changes, the
+// persistent FSM state can behave as a complete snapshot, and be able to recover
+// without an external snapshot just for persisting the raft configuration.
+type ConfigurationStore interface {
```

Comment:
Random drive-by comment: if this interface is only useful for FSMs, should you embed the FSM interface in here as well?

I tend to find that to be a fairly self-documenting behavior I appreciate.

Reply:
Good idea, i'll add that